### PR TITLE
SQS Cleanup, and fix #1105

### LINKED ISSUE #1105: SQS Fifo support
Hi!
Currently I'm working on a Service that pops messages from a sqs fifo queue.
In my integration test, I am mocking the AWS services like this: 

```
   @mock_s3
    @mock_sqs
    @patch('redis.StrictRedis')
    @patch('new_backend.resolvers.api.get_http_client')
    def test_complete_backend(self, get_http_client, redis_class_mock):
        os.environ['AWS_REGION'] = "eu-west-1"
        os.environ['AWS_SQS_QUEUE_NAME'] = "test.fifo"
        os.environ['AWS_S3_BUCKET_NAME'] = "inbound-smtp-dev-ireland"
        sqs = boto3.resource('sqs', region_name=os.environ['AWS_REGION'])
        sqs_queue = sqs.create_queue(QueueName=os.environ['AWS_SQS_QUEUE_NAME'], Attributes={'FifoQueue':'true'})

        #sqs_queue = sqs.get_queue_by_name(QueueName=os.environ['AWS_SQS_QUEUE_NAME'])
        sqs_queue.should_not.be.none
        assert sqs_queue.attributes.get('FifoQueue') == 'true'
        sqs_queue.should.have.property('url').should.contain(os.environ['AWS_SQS_QUEUE_NAME'])
        s3_client = boto3.client('s3', region_name=os.environ['AWS_REGION'])

        s3_connection = boto3.resource('s3', region_name=os.environ['AWS_REGION'])
        s3_bucket = s3_connection.create_bucket(Bucket=os.environ['AWS_S3_BUCKET_NAME'])
```

In my understanding, the assert in line 13 should not fail. But it does. The attribute FifoQueue is not set neither exists. However, if I run the test without mocks, directly on AWS services, the attribute exists and is true.

Is moto (my version is 1.1.1) not supporting Fifo queues yet? Or am I missing something?

Thank you and have a nice week!

### PR DESCRIPTION
**Pull request by terrycain**
Various cleanups in the SQS part. If [this](https://github.com/spulec/freezegun/pull/210) PR is merged, can then cut down test time from 15s to 9s and could probably get test_messages_polling down from 3s.

Passed more options to Queue class.
Cleaned some parts of the web request attribute management.

Fixes #1105

### CODE REVIEW

**Comment by terrycain** [comment]
At last :smile:

File: moto/sqs/responses.py [hunk 1]
```diff
@@ -28,8 +28,7 @@ def sqs_backend(self):
     @property
     def attribute(self):
         if not hasattr(self, '_attribute'):
-            self._attribute = dict([(a['name'], a['value'])
-                                    for a in self._get_list_prefix('Attribute')])
+            self._attribute = self._get_map_prefix('Attribute', key_end='Name', value_end='Value')
```
Comment (JackDanger) [inline moto/sqs/responses.py]: Oooh, nice

**Comment by JackDanger** [comment]
This is great!

**Comment by JackDanger** [comment]
This is now released as Moto 1.1.14

**Pull request merged**